src/prefect/tasks/gcp/bigquery.py

```diff
@@ -382,6 +382,10 @@ def run(
         load_job = client.load_table_from_uri(uri, table_ref, job_config=job_config)
         load_job.result()  # block until job is finished
 
+        # remove unpickleable attributes
+        load_job._client = None
+        load_job._completion_lock = None
```

Comment:
I'm a bit wary of modifying private attributes on objects in non-prefect libraries. If the implementation in the bigquery library changes, this may no longer work, and it may also lead to errors for users relying on methods in the `load_job` object. Perhaps pickling support should be fixed upstream in the bigquery library instead?

Reply:
As you say, this is a temporary patch and these methods are relying on the `_client` and `_completion_lock` attribute.
- `exists`
- `reload`
- `cancel`

> Perhaps pickling support should be fixed upstream in the bigquery library instead?

Hmm, I think prefect is responsible for serialization, not bigquery library.
If we want to fix it correctly, I think prefect should define an interface of pickleable return value instead of the LoadJob object itself.

Reply:
> Hmm, I think prefect is responsible for serialization, not bigquery library.

We use `pickle`, which I'd argue most objects in Python should support. I think an issue report to the upstream bigquery library for supporting pickling here would be useful and valid.

That said, the fix here seems fine for now.